cmds/core/yes/yes_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
+	cmd := testutil.Command(t)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		t.Logf("Could not return a pipe that will be connected to the command's standard output, received err: %v", err)
+	}
+
+	if err := cmd.Start(); err != nil {
+		t.Log(err)
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 1000; i++ {
+		if in.Scan() && in.Text() != "y" {
+			t.Errorf("Expected a 'y', but received %v", in.Text())
+			break
+		}
+	}
+	if err := cmd.Process.Kill(); err != nil {
+		t.Errorf("Killing the child process: received %v, but wanted nil", err)
+	}
+}
+
+func TestYesArgs(t *testing.T) {
```

Comment:
As above. I would actually suggest you compress these two tests into one test that can call yes with various arguments. Right now almost all the test logic is duplicated beyond the args and the string. One example to look at is pkg/ebda/ebda_test.go. Look at how we can test the same functionality with different arguments or setup. You could do something similar

Reply:
gan, I agree, but can we get it working and then merge?

Reply:
I just pushed my recent updates. The code works. If there are any other updates needed to be made, I will go ahead and merge my two test functions together :)! Thank you.